test/unit/log.provider.test.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {sinon} from '@loopback/testlab';
+import {ParsedRequest} from '@loopback/core';
+import {LogProvider} from '../..';
+
+describe('LogProvider', () => {
+  // tslint:disable-next-line:no-any
+  let spy: any;
+  beforeEach(() => {
+    spy = sinon.spy(console, 'log');
+  });
+
+  afterEach(() => {
+    spy.restore();
+  });
+
+  it('logs a value without a start time', () => {
+    const match = '/test : () => test message';
+
+    runTest(match);
```

Comment:
I find this test difficult to read, because it's not clear how is the logger executed and what exactly is being tested.

Here is what I consider as a better version (please forgive possible mistakes, I am typing from my head):

```ts
const req = <ParsedRequest> {url: '/test'};

let log: LogFn;
beforeEach(givenLogger);

it('logs a value without a start time', () => {
  log(req, [], 'test message');
  sinon.assert.calledWith(spy, sinon.match('/test : () => test message');
}

function givenLogger() {
  log = new LogProvider().value();
}
```

BTW, I believe testlab has integration between sinon and should.js configured, thus you should be able to do something along the lines of `expect(stub).to.be.calledWith(sinon.match(...))`

Reply:
I wasn't able to use `expect` with a spy. I did:
```
expect(spy).to.be.calledWith(sinon.match())
```
but `.calledWith` needs the spy/object to be matched against as an arg. Hence leaving it as an assertion.

Reply:
> `expect(spy).to.be.calledWith(sinon.match())`

Hmm, this looks like a bug in testlab to me. I did a quick search, we are using `should/as-function` for `expect`, which has different `Assertion` class/constructor than regular `should`. Now `should-sinon` modifies only the `Assertion` prototype from `should`, not from `should/as-function`, therefore it does not add sinon-related assertions to our `expect.

Let me fill an issue for that: https://github.com/strongloop/loopback-next/issues/641